lib/next_ls/commands/from_pipe.ex

```diff
@@ -0,0 +1,116 @@
+defmodule NextLS.Commands.FromPipe do
+  @moduledoc false
+  alias GenLSP.Enumerations.ErrorCodes
+  alias GenLSP.Structures.Position
+  alias GenLSP.Structures.Range
+  alias GenLSP.Structures.TextEdit
+  alias GenLSP.Structures.WorkspaceEdit
+
+  defp opts do
+    Schematic.map(%{
+      position: Schematic.map(%{character: Schematic.int(), line: Schematic.int()}),
+      uri: Schematic.str(),
+      text: Schematic.list(Schematic.str())
+    })
+  end
+
+  def new(opts) do
+    {:ok, %{text: text, uri: uri, position: position}} = Schematic.unify(opts(), Map.new(opts))
```

Comment:
I was going to suggest wrapping the whole handle_event callback in the NextLS module in a try/rescue, so that we can log an error to the user and not crash the whole server.

You can publish a window/showMessage notification (I think there are examples in the codebase), so that it actually presents an UI notification to the user's editor